cedar-drt/fuzz/src/schema.rs

```diff
@@ -1164,8 +1191,21 @@ impl Schema {
             SchemaTypeVariant::Boolean => {
                 self.arbitrary_value_for_type(&Type::bool(), hierarchy, max_depth, u)
             }
-            SchemaTypeVariant::Long => {
-                self.arbitrary_value_for_type(&Type::long(), hierarchy, max_depth, u)
+            SchemaTypeVariant::Long { min_opt, max_opt } => {
+                match (min_opt, max_opt) {
+                    (None, None) => {
+                        self.arbitrary_value_for_type(&Type::long(), hierarchy, max_depth, u)
+                    }
+                    (Some(min), Some(max)) =>
+                    // TODO: We should probably prefer a Long from the pool if it is in range.
+                    {
+                        Ok(Value::Lit(u.int_in_range(*min..=*max)?.into()))
+                    }
+                    // TODO: This is messy. Find a way to restructure
+                    // SchemaTypeVariant so it has the deserialization behavior
+                    // we want without pushing this mess onto all consumers.
+                    _ => panic!("Corrupt SchemaTypeVariant::Long"),
```

Comment:
It looks like you probably want `(min_opt, max_opt)` to be an optional pair, so that you're guaranteed to have both or neither.